File: server/src/main/java/com/productivityapp/demo/domain/dto/TodoDto.java
package com.productivityapp.demo.domain.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.productivityapp.demo.domain.enumeration.Priority;

import java.time.LocalDate;

public record TodoDto(

        String title,

        String description,

        @JsonFormat(pattern = "dd/MM/yyyy")
        LocalDate publishDate,

        @JsonFormat(pattern = "dd/MM/yyyy")
        LocalDate deadLineDate,

        Priority priority,

        boolean isDone
) {
}

File: client/src/app/dtos/todo-dto.ts
import { Priority } from "../enums/priority";

export class TodoDto {

    id: number;
    title: string;
    description: string;
    publishDate : Date;
    deadLineDate : Date;
    priority : Priority;
    isDone: boolean;

    constructor(id:number,title: string,description: string,publishDate : Date,deadLineDate : Date,priority : Priority,isDone: boolean){
    this.id = id;
    this.title = title ;
    this.description = description ;
    this.publishDate = publishDate;
    this.deadLineDate = deadLineDate ;
    this.priority = priority ;
    this.isDone = isDone 
    }
}